File: in_java/Pattern_22.java
package in_java;

public class Pattern_22 extends Pattern {
    public Pattern_22(int row, int column, String symbol) {
        super(row, column, symbol);
    }

    public void print() {
        int firstSymbol;
        int secondSymbol;
        int halfColumn = this.column / 2;

        for (int i = 0; i < this.row; i++) {
            firstSymbol = 1;
            secondSymbol = 65;

            for (int j = 0; j < this.column; j++) {
                if (j >= halfColumn - i && j <= halfColumn + i) {
                    if (j <= halfColumn) {
                        System.out.print(firstSymbol);
                        firstSymbol++;
                    } else {
                        System.out.print((char) secondSymbol);
                        secondSymbol++;
                    }
                } else {
                    System.out.print(" ");
                }
            }
            System.out.println();
        }
    }
}

File: in_rust/src/pattern_22.rs
pub fn pattern_22() {
    let row = 5;
    let column = 9;
    let half_column = column / 2;

    let mut first_symbol: u8;
    let mut second_symbol: u8;

    for i in 0..row {
        first_symbol = 1;
        second_symbol = 65;

        for j in 0..column {
            if j >= half_column - i && j <= half_column + i {
                if j <= half_column {
                    print!("{}", first_symbol);
                    first_symbol += 1;
                } else {
                    print!("{}", second_symbol as char);
                    second_symbol += 1;
                }
            } else {
                print!(" ");
            }
        }
        println!();
    }
}
